Lib/webbrowser.py

```diff
@@ -15,14 +15,15 @@ class Error(Exception):
 
 _browsers = {}          # Dictionary of available browser controllers
 _tryorder = []          # Preference order of available browsers
+_preferred_browser = None # The preferred browser for the current environment
```

Comment:
Rather than only allowing a single preferred browser, perhaps it would make sense to make this a `_preferred_browsers` set? That would have a ripple effect on the other updates below.